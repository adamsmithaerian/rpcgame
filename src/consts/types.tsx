export enum Shape {
  Rock,
  Paper,
  Scissors
}

export enum Result {
  Win,
  Lose,
  Tie
}

export const shapeNames: Array<String> = ['Rock', 'Paper', 'Scissors'];

export const resultNames: Array<String> = ['Win', 'Lose', 'Tie'];
