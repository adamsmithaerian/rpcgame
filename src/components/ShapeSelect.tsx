import * as React from 'react';
import { Shape, shapeNames } from '../consts/types';

import './ShapeSelectForm.css';

interface SelectProps {
  onChange: (value: Shape) => void;
  value: Shape;
}

class ShapeSelect extends React.Component<SelectProps, any> {
  render() {
    return (
      <div className="Shape-select">
        <select
          onChange={(e: any) =>
            this.props.onChange(parseInt(e.target.value, 10))
          }
          value={this.props.value}
        >
          <option value={Shape.Rock}>{shapeNames[Shape.Rock]}</option>
          <option value={Shape.Paper}>{shapeNames[Shape.Paper]}</option>
          <option value={Shape.Scissors}>{shapeNames[Shape.Scissors]}</option>
        </select>
      </div>
    );
  }
}

export default ShapeSelect;
