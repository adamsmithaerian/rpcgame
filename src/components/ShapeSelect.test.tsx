import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import ShapeSelect from './ShapeSelect';
import { Shape } from '../consts/types';

describe('ShapeSelect component should', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(
      <ShapeSelect 
        onChange={() => {}}
        value={Shape.Rock}
      />,
      div
    );
  });

  it('match snapshot', () => {
    const tree = renderer
      .create(
        <ShapeSelect
          onChange={() => {}}
          value={Shape.Rock}
        />
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });  
});

