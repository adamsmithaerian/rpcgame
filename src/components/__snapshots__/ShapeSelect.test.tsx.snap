// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPannel component should match snapshot 1`] = `
<div
  className="Shape-select"
>
  <select
    onChange={[Function]}
    value={0}
  >
    <option
      value={0}
    >
      Rock
    </option>
    <option
      value={1}
    >
      Paper
    </option>
    <option
      value={2}
    >
      Scissors
    </option>
  </select>
</div>
`;

exports[`Shape Select component should match snapshot 1`] = `
<div
  className="Shape-select"
>
  <select
    onChange={[Function]}
    value={0}
  >
    <option
      value={0}
    >
      Rock
    </option>
    <option
      value={1}
    >
      Paper
    </option>
    <option
      value={2}
    >
      Scissors
    </option>
  </select>
</div>
`;

exports[`ShapeSelect component should match snapshot 1`] = `
<div
  className="Shape-select"
>
  <select
    onChange={[Function]}
    value={0}
  >
    <option
      value={0}
    >
      Rock
    </option>
    <option
      value={1}
    >
      Paper
    </option>
    <option
      value={2}
    >
      Scissors
    </option>
  </select>
</div>
`;
